pkg/controller/blobstorage/blobstorage_controller.go

```diff
@@ -146,5 +176,34 @@ func (r *ReconcileBlobStorage) Reconcile(request reconcile.Request) (reconcile.R
 		}
 		return reconcile.Result{Requeue: true, RequeueAfter: time.Second * 30}, nil
 	}
+
+	// unsupported strategy
+	if err = r.updatePhaseFailed(instance, "unsupported deployment strategy"); err != nil {
+		return reconcile.Result{}, errorUtil.Wrapf(err, "failed to update instance %s in namespace %s", instance.Name, instance.Namespace)
+	}
 	return reconcile.Result{}, errorUtil.New(fmt.Sprintf("unsupported deployment strategy %s", stratMap.BlobStorage))
 }
+
+func (r *ReconcileBlobStorage) updatePhaseFailed(instance *v1alpha1.BlobStorage, msg v1alpha1.StatusMessage) error {
+	instance.Status.Phase = v1alpha1.PhaseFailed
+	instance.Status.Message = msg
+	return r.client.Status().Update(r.ctx, instance)
+}
+
+func (r *ReconcileBlobStorage) updatePhaseComplete(instance *v1alpha1.BlobStorage, msg v1alpha1.StatusMessage) error {
+	instance.Status.Phase = v1alpha1.PhaseComplete
+	instance.Status.Message = msg
+	return r.client.Status().Update(r.ctx, instance)
+}
+
+func (r *ReconcileBlobStorage) updatePhaseInProgress(instance *v1alpha1.BlobStorage, msg v1alpha1.StatusMessage) error {
+	instance.Status.Phase = v1alpha1.PhaseInProgress
+	instance.Status.Message = msg
+	return r.client.Status().Update(r.ctx, instance)
+}
+
+func (r *ReconcileBlobStorage) updatePhaseDeletionInProgress(instance *v1alpha1.BlobStorage, msg v1alpha1.StatusMessage) error {
+	instance.Status.Phase = v1alpha1.PhaseDeleteInProgress
+	instance.Status.Message = msg
+	return r.client.Status().Update(r.ctx, instance)
+}
```

Comment:
Although it's only three lines I'd be tempted to de-duplicate these four functions into.

```
func (r *ReconcileBlobStorage) updatePhaseInProgress(inst *v1alpha1.BlobStorage, msg v1alpha1.StatusMessage) error {
    return updatePhase(inst, v1alpha1.PhaseInProgress, msg)
}

...

func (r *ReconcileBlobStorage) updatePhase(inst *v1alpha1.BlobStorage, phase v1alpha1.StatusPhase, msg v1alpha1.StatusMessage) error {
    inst.Status.Phase = phase
    inst.Status.Message = msg
    if err := r.client.Status().Update(r.ctx, inst); err != nil {
        return errors.Wrap(err, "failed to update resource status phase and message")
    }
    return nil
}
```

This'd allow us to add a bit more content into the main body, like wrapping the error, without having to repeat ourselves.

I'm also not overly familiar with it, but we may be able to use the `runtime` `Field` and `SetField` functions for making this generic, instead of specific to `BlobStorage`.